runtime/vm/createramclass.cpp

```diff
@@ -2818,7 +2820,10 @@ internalCreateRAMClassFromROMClass(J9VMThread *vmThread, J9ClassLoader *classLoa
 					module = javaVM->unamedModuleForSystemLoader;
 				}
 			} else {
-				if ((LOAD_LOCATION_PATCH_PATH == locationType) || (LOAD_LOCATION_MODULE == locationType)) {
+				if (J9ROMCLASS_IS_PRIMITIVE_TYPE(romClass)
```

Comment:
This particular code pre-emtively assigns the J9Class to the java.base J9Module. At this point it is not constructed, but when java.base is created it will be completed. Since the J9Class's already point to it they wont have to be updated.

Same applies for the bootloader unnamed module

Reply:
Thanks for clarifying.  I had to check the code that initializes `javaBaseModule` as I thought it was done latter.

This looks fine to me.